cfme/configure/settings.py

```diff
@@ -1,173 +1,223 @@
+import attr
 import re
-from navmazing import NavigateToAttribute, NavigateToSibling
-from widgetastic.utils import Version, VersionPick
-from widgetastic.widget import View
-from widgetastic_patternfly import (BootstrapSwitch,
-                                    Input, Button, CheckableBootstrapTreeview, Dropdown)
 
-from cfme.base.login import BaseLoggedInPage
+from cfme.base import BaseEntity, BaseCollection
 from cfme.base.ui import MySettingsView
+from cfme.base.login import BaseLoggedInPage
 from cfme.utils.appliance import Navigatable
 from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
 from cfme.utils.pretty import Pretty
 from cfme.utils.update import Updateable
+from navmazing import NavigateToAttribute, NavigateToSibling
 from widgetastic_manageiq import Table, BootstrapSelect, BreadCrumb, Text, ViewButtonGroup
+from widgetastic_patternfly import (BootstrapSwitch,
+                                    Input, Button, CheckableBootstrapTreeview, Dropdown)
+from widgetastic.utils import Version, VersionPick
+from widgetastic.widget import View
 
 
-class TimeProfileAddForm(View):
+class TimeProfileForm(View):
     description = Input(id='description')
     scope = BootstrapSelect('profile_type')
     timezone = BootstrapSelect('profile_tz')
     days = BootstrapSwitch(name='all_days')
     hours = BootstrapSwitch(name='all_hours')
-    save_button = Button(VersionPick({Version.lowest(): 'Save', '5.9': 'Add'}))
-    configuration = Dropdown('Configuration')
-    table = Table("//div[@id='main_div']//table")
-    save_edit_button = Button('Save')
-    cancel_button = Button('Cancel')
-    help_block = Text("//span[contains(@class, 'help-block')]")
+    cancel = Button('Cancel')
+    help_block = Text('//span[contains(@class, "help-block")]')
 
 
-class TimeprofileAddEntities(View):
+class TimeProfileEntities(View):
+    table = Table('//div[@id="main_div"]//table')
     breadcrumb = BreadCrumb()
     title = Text('//div[@id="main-content"]//h3')
 
 
-class TimeProfileAddFormView(BaseLoggedInPage):
-    timeprofile_form = View.nested(TimeProfileAddForm)
-    entities = View.nested(TimeprofileAddEntities)
+class TimeProfileView(BaseLoggedInPage):
+    form = View.nested(TimeProfileForm)
```

Comment:
@anurag03 @psav 
I see some of the other comments concerning view structure on this PR. This does not look quite right to me as is, and hopefully is not against what @psav has requested.

Form should not be a view nested under the common TimeProfileView, because there is only a form section on the Add and Edit pages.

The `TimeProfileForm` class should be inherited by views nested under the `TimeProfileAddView` and `TimeProfileEditView`

These nested views should look like:
```python
class TimeProfileAddView(TimeProfileView):
    @View.nested
    def form(TimeProfileForm):
        add = Button(blah-blah-blah)
```

This way the nested form view is only available in Add/Edit views, and isn't part of the `TimeProfileAllView` - which doesn't have a form.